improve docs/README.md

christo4ferris opened the pull request:
## Description

Refactored the README to be better organized and a bit more than just a TOC. Eliminated some redundant information, and eliminated TOC entries without a link.
## Motivation and Context

The documentation needs some improvement to make it more approachable. There is more to be done, but this is a start down that path.

Fixes #
## How Has This Been Tested?

verified all links in my repo.
## Checklist:

<!-- To check a box, and an 'x': [x] -->

<!-- To uncheck box, add a space: [ ] -->

<!-- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] I have added a [Signed-off-by](https://github.com/hyperledger/fabric/blob/master/CONTRIBUTING.md#legal-stuff).
- [x] I have either added documentation to cover my changes or this change requires no new documentation.
- [x] I have either added unit tests to cover my changes or this change requires no new tests.
- [x] I have run [golint](https://github.com/golang/lint) and have fixed valid warnings in code I have added or modified. This tool generates false positives so you may choose to ignore some warnings. The goal is clean, consistent, and readable code.

<!-- The continuous integration build process will run [make checks](https://github.com/hyperledger/fabric/blob/master/Makefile#L22) to confirm that tests pass and that code quality meets minimum standards. You may optionally run this locally as PRs will not be accepted until they pass. -->

Signed-off-by: Christopher Ferris chrisfer@us.ibm.com

christo4ferris committed:
improve docs/README.md

refactor the README.md under docs to be more organized and more
informative.

Signed-off-by: Christopher Ferris <chrisfer@us.ibm.com>

[ci skip]

christo4ferris commented:
reverify Jenkins

gabre commented on docs/README.md:
```diff
@@ -1,117 +1,67 @@
-     - Chaincode Execution
+## Chaincode developer guide:
+
+- [Setting up the development environment](dev-setup/devenv.md) - when developing and testing chaincode, or an application that leverages the fabric API or SDK, you'll probably want to run the fabric locally on your laptop to test. You can use the same set up that Fabric developers use.
```
I think there is a small typo here: "the same set up" -> "the same setup"

christo4ferris replied:
fixed

christo4ferris committed:
fix typo

s/same set up/same setup/

Signed-off-by: Christopher Ferris <chrisfer@us.ibm.com>

christo4ferris commented:
reverify Jenkins

christo4ferris committed:
adapted changes from #2132

adapted changes from #2132 because that PR conflicts with this one.

fixed links in mkdocs rendering. added link to LICENSE. fixed error
with markdown extensions in mkdocs.yml

Signed-off-by: Christopher Ferris <chrisfer@us.ibm.com>

srderson commented on docs/README.md:
```diff
@@ -1,117 +1,67 @@
 
-Welcome to the Linux Foundation Hyperledger Project documentation README. This page contains: 
+If you are new to the project, you can begin by reviewing the following links. If you'd prefer to dive right in, see the [Quickstart](#quickstart-documentation) section, below.
+- [Whitepaper WG](https://github.com/hyperledger/hyperledger/wiki/Whitepaper-WG): where the community is developing a whitepaper to explain the motivation and goals for the project
```
Period usage seems to differ between link descriptions. Some contain a period, others do not. Also, some links are followed by ': ' where others are followed by ' - '. Otherwise LGTM

christo4ferris replied:
I'll clean that up @srderson thanks

christo4ferris committed:
fix punctuation consistency

fix punctuation consistency

Signed-off-by: Christopher Ferris <chrisfer@us.ibm.com>

srderson commented:
LGTM
